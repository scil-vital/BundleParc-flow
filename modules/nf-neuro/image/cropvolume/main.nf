
process IMAGE_CROPVOLUME {
    tag "$meta.id"
    label 'process_single'

    container "scilus/scilpy:2.2.0_cpu"

    input:
    tuple val(meta), path(image), path(bounding_box)

    output:
    tuple val(meta), path("*_cropped.nii.gz"), emit: image
    tuple val(meta), path("*.pkl")           , emit: bounding_box, optional: true
    path "versions.yml"                      , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"

    def input_bbox = bounding_box ? "--input_bbox $bounding_box" : ""
    def suffix = task.ext.first_suffix ? "${task.ext.first_suffix}_cropped" : "cropped"
    def output_bbox = task.ext.output_bbox ? "--output_bbox ${prefix}_${suffix}_bbox.pkl" : ""

    """
    scil_volume_crop $image ${prefix}_${suffix}.nii.gz $input_bbox $output_bbox

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def suffix = task.ext.first_suffix ? "${task.ext.first_suffix}_cropped" : "cropped"

    """
    scil_volume_crop -h

    touch ${prefix}_${suffix}.nii.gz

    if $task.ext.output_bbox;
    then
        touch ${prefix}_${suffix}_bbox.pkl
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
