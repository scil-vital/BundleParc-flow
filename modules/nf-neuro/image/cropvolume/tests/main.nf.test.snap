{
    "image - cropvolume - stub-run": {
        "content": [
            [
                "versions.yml:md5,da695528b3fb62e539bfd19768c0267a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T20:03:48.863431786"
    },
    "image - cropvolume": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,93cc10f69b0829409c69bf743e9c95ef"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,da695528b3fb62e539bfd19768c0267a"
                ],
                "bounding_box": [
                    
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,93cc10f69b0829409c69bf743e9c95ef"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,da695528b3fb62e539bfd19768c0267a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T20:03:12.376086542"
    },
    "image - cropvolume - outputbbox": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,93cc10f69b0829409c69bf743e9c95ef"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped_bbox.pkl:md5,bc8e962d194d3ebdb93bfeb1e318d5b8"
                    ]
                ],
                "2": [
                    "versions.yml:md5,da695528b3fb62e539bfd19768c0267a"
                ],
                "bounding_box": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped_bbox.pkl:md5,bc8e962d194d3ebdb93bfeb1e318d5b8"
                    ]
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,93cc10f69b0829409c69bf743e9c95ef"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,da695528b3fb62e539bfd19768c0267a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T20:03:30.051801322"
    }
}